WebContent/zip-ext.js

```diff
@@ -208,11 +208,165 @@
 	FileWriter.prototype = new Writer();
 	FileWriter.prototype.constructor = FileWriter;
 
+	function DBWriter(contentType) {
+		var db, tempDB, that = this, blobs, dbName = "zipjs", instance;
+
+		function init(callback, onerror) {
+			var request = indexedDB.open(dbName, 5);
+			request.onerror = onerror;
+			request.onupgradeneeded = function (event) {
+				db = event.target.result;
+				db.createObjectStore("instances", { autoIncrement: true });
+			};
+			request.onsuccess = function (event) {
+				db = event.target.result;
+				addInstance(callback, onerror);
+			};
+			blobs = [];
+		}
+
+		function addInstance(callback, onerror) {
+			var t = db.transaction(["instances"], "readwrite"),
+				request = t.objectStore("instances").put(Date.now());
+			request.onerror = onerror;
+			request.onsuccess = function (event) {
+				instance = request.result;
+				cleanUpOldInstances();
+				window.addEventListener("storage", pongDB);
+				window.addEventListener("unload", close);
+				addInstanceDB(callback, onerror);
+			};
+		}
+
+		function addInstanceDB(callback, onerror) {
+			var request;
+			try {
+				request = indexedDB.open(dbName + "_" + instance, {temporary: true});
+			} catch (e) {
+				request = indexedDB.open(dbName + "_" + instance);
+			}
+			request.onerror = onerror;
+			request.onupgradeneeded = function (event) {
+				tempDB = event.target.result;
+				tempDB.createObjectStore("files", { autoIncrement: true });
+			};
+			request.onsuccess = function (event) {
+				callback();
+			};
+		}
+
+		function close() {
+			blobs = [];
```

Comment:
Replace `blobs = undefined;` with `blobs = null;`. The purpose of `close` is to destroy the state of the writer, not to reset/reinitialize it. Assigning `null` causes `writeUint8Array` (below) to throw a TypeError when the callback is called again.